File: FlutterApp/lib/models/trade_models.dart
import 'dart:convert';

class TradeSignal {
  final int id;
  final String symbol;
  final String direction;
  final double entry;
  final double tp1;
  final double sl;
  final String timeframe;
  final String createdAt;

  TradeSignal({
    required this.id,
    required this.symbol,
    required this.direction,
    required this.entry,
    required this.tp1,
    required this.sl,
    required this.timeframe,
    required this.createdAt,
  });

  factory TradeSignal.fromJson(Map<String, dynamic> json) {
    return TradeSignal(
      id: json['id'] as int,
      symbol: json['symbol'] as String,
      direction: json['direction'] as String,
      entry: json['entry']?.toDouble() ?? 0.0,
      tp1: json['tp']?.toDouble() ?? 0.0,
      sl: json['sl']?.toDouble() ?? 0.0,
      timeframe: json['timeframe'] as String,
      createdAt: json['created_at'] as String,
    );
  }
}

class MarketCandle {
  final DateTime date;
  final double open;
  final double high;
  final double low;
  final double close;
  final double volume;

  MarketCandle({
    required this.date,
    required this.open,
    required this.high,
    required this.low,
    required this.close,
    required this.volume,
  });
}

File: Models/TradeModels.swift
import Foundation

enum TradeDirection: String, Codable {
    case long = "LONG"
    case short = "SHORT"
}

struct TradeSignal: Identifiable, Codable {
    var id = UUID()
    let symbol: String
    let direction: TradeDirection
    let entry: Double
    let stopLoss: Double
    let takeProfit1: Double
    let takeProfit2: Double?
    let score: Double
    let pattern: String
    let timestamp: Date
    
    // Institutional Metadata
    var isBackendValidated: Bool = false
    var confluence: String? = nil
    var aiRationale: String? = nil
    
    enum CodingKeys: String, CodingKey {
        case symbol, direction, entry, score, pattern, timestamp
        case stopLoss = "sl"
        case takeProfit1 = "tp"
        case takeProfit2 = "tp2"
        case isBackendValidated = "validated"
        case confluence
        case aiRationale = "ai_rationale"
    }
}

struct MarketCandle: Identifiable, Codable {
    var id: Double { timestamp }
    let timestamp: Double
    let open: Double
    let high: Double
    let low: Double
    let close: Double
    let volume: Double
}

struct Position: Identifiable, Codable {
    var id: String { "\(symbol)-\(direction)" }
    let symbol: String
    let entryPrice: Double
    let markPrice: Double
    let size: Double
    let pnl: Double
    let direction: TradeDirection
    var broker: String? = nil // To support grouping
}

struct AccountSummary: Codable {
    let equity: Double
    let balance: Double
    let available: Double
    let openPositions: Int
}

struct BrokerAccount: Identifiable {
    let id: String // Broker Name (e.g. Binance, Exness)
    let summary: AccountSummary
    let positions: [Position]
}

struct EquityPoint: Codable, Identifiable {
    var id: String { time }
    let equity: Double
    let time: String
}

struct LeaderboardStat: Codable, Identifiable {
    var id: String { pattern }
    let pattern: String
    let total: Int
    let wins: Int
    let losses: Int
    let win_rate: Double
}
